src/thor/triplegbuilder.cc

```diff
@@ -1340,7 +1340,7 @@ TripLegBuilder::Build(const AttributesController& controller,
       }
     }
     // Update elapsed time at the end of the edge, store this at the next node.
-    elapsedtime += edge_itr->elapsed_time * edge_pct;
+    elapsedtime = edge_itr->elapsed_time * edge_pct;
```

Comment:
we update the last node's elapsed time in the end, and that should be a accumulation of all the edges that the loop iterate through. Maybe we should add another variable here to cache the sum and add it to the last node

Reply:
youre correct im working on a more complex fix based on the pseudo code i shared a few days ago